Kudu.Core/Deployment/Generator/ExternalCommandBuilder.cs

```diff
@@ -85,5 +114,94 @@ protected void RunCommand(DeploymentContext context, string command)
                 throw;
             }
         }
+
+        /// <summary>
+        /// Mark public for testing
+        /// </summary>
+        public IList<string> GetPostBuildActionScripts()
+        {
+            List<string> scriptFilesGroupedAndSorted = new List<string>();
+            List<string> scriptFolders = new List<string>();
+
+            // "/site/deployments/tools/PostDeploymentActions" (can override with %POST_DEPLOYMENT_ACTIONS_DIR%)
+            var defaultPath = Path.Combine(Environment.DeploymentToolsPath, PostDeploymentActions);
+            if (!string.IsNullOrEmpty(System.Environment.GetEnvironmentVariable(WellKnownEnvironmentVariables.PostDeploymentActionsDirectoryKey)))
+            {
+                defaultPath = System.Environment.GetEnvironmentVariable(WellKnownEnvironmentVariables.PostDeploymentActionsDirectoryKey);
+            }
+
+            if (FileSystemHelpers.DirectoryExists(defaultPath))
+            {
+                scriptFolders.Add(defaultPath);
+            }
+
+            // D:\Program Files (x86)\SiteExtensions
+            string siteExtensionFolder = Path.Combine(System.Environment.GetFolderPath(System.Environment.SpecialFolder.ProgramFilesX86), "SiteExtensions");
+            // Find all enviroment variable with format {package id}_EXTENSION_VERSION,
+            // And find scripts from PostDeploymentActions" folder
+            foreach (DictionaryEntry entry in System.Environment.GetEnvironmentVariables())
+            {
+                string key = entry.Key.ToString();
+                string val = entry.Value.ToString();
+                NuGetVersion ver = null;
+                if (key.EndsWith("_EXTENSION_VERSION", StringComparison.Ordinal)
+                    && (string.Equals("latest", val, StringComparison.OrdinalIgnoreCase) || NuGetVersion.TryParse(val, out ver)))   // either latest or a specific version
+                {
+                    if (ver != null)
+                    {
+                        // "D:\Program Files (x86)\SiteExtensions", "{PackageName}", "{version}", "PostDeploymentActions"
+                        //  => D:\Program Files (x86)\SiteExtensions\{PackageName}\{version}\PostDeploymentActions
+                        string scriptFolder = Path.Combine(siteExtensionFolder, key.Replace("_EXTENSION_VERSION", string.Empty), ver.ToNormalizedString(), PostDeploymentActions);
```

Comment:
Or better yet, the whole extension path. And note that 'extension' is a more correct term than 'package'